src/handlers/motions/motionFinalized/helpers.ts

```diff
@@ -49,3 +52,48 @@ export const getStakerReward = async (
     isClaimed: false,
   };
 };
+
+export const getParsedActionFromMotion = async (
+  motionAction: string,
+  colonyAddress: string,
+): Promise<TransactionDescription | undefined> => {
+  const colonyClient = await networkClient.getColonyClient(colonyAddress);
+
+  // We are only parsing this action in order to know if it's a add/edit domain motion. Therefore, we shouldn't need to try with any other client.
+  try {
+    return colonyClient.interface.parseTransaction({
+      data: motionAction,
+    });    
+  } catch {
+    verbose(`Unable to parse ${motionAction} using colony client`);
+    return undefined;
+  }
+};
+
+export const linkPendingDomainMetadataWithDomain = async (action: string, colonyAddress: string, finalizedMotion: MotionQuery) => {
+  const parsedDomainAction = await getParsedActionFromMotion(action, colonyAddress);
+  if (parsedDomainAction?.name === ColonyOperations.AddDomain) {
+    const colonyClient = await networkClient.getColonyClient(colonyAddress);
+    const domainCount = await colonyClient.getDomainCount();
+    // The new domain should be created by now, so we just get the total of existing domains
+    // and use that as an id to link the pending metadata.
+    const nativeDomainId = domainCount.toNumber();
```

Comment:
I thought MotionFinalized happens before the domain is created, is it after? Or is there some race conditions that tends to win on this side?

Reply:
The user clicks the Finalize button, then the on-chain finalizeMotion call will execute the action, and then the MotionFinalized event will be emitted and picked up by the block ingestor, by which point the domain should/will have been created on-chain